### PR TITLE
Adopt OCaml's code of conduct

### PR DESCRIPTION
**Pull request by balat**
This PR proposes to add the OCaml Code of Conduct. Full text is available at https://github.com/ocaml/code-of-conduct, and a discussion around it is at https://discuss.ocaml.org/t/ocaml-community-code-of-conduct/10494.

### CODE REVIEW

**Comment by raphael-proust** [comment]
We should close https://github.com/ocsigen/lwt/pull/982

**Comment by balat** [comment]
Oh I didn't see it. Yes accept one or the other.

**Comment by raphael-proust** [comment]
Yes we probably need to advertise the list of people that receive these emails.

@balat has sent an email on there confirming that everyone on there is ok being on there. I am and I am. But let's wait until everyone has accepted to be the recipient of those emails before we list them here.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Code of Conduct
+
+This project has adopted the [OCaml Code of Conduct](https://github.com/ocaml/code-of-conduct/blob/main/CODE_OF_CONDUCT.md).
+
+# Enforcement
+
+This project follows the OCaml Code of Conduct [enforcement policy](https://github.com/ocaml/code-of-conduct/blob/main/CODE_OF_CONDUCT.md#enforcement).
+
+To report any violations, please contact Jérôme
+Vouillon, Raphael Proust, Vincent Balat, Hugo Heuzard and Gabriel
```
Comment (raphael-proust) [inline CODE_OF_CONDUCT.md:10]:
```suggestion
Vouillon, Raphaël Proust, Vincent Balat, Hugo Heuzard and Gabriel
```

**Pull request merged**